### ISSUE TITLE
Google Smart Chip Phase 3 - Roll out

### ISSUE THREAD

**Issue by saumier** [culturecreates/artsdata-google-workspace-smart-chip#9]
Prepare the release of the smart-chip ONLY and not the code for the extension to search names for reconiliation.  This project shifted away from using Google Sheets for reconciling in favour of using Artsdata with the Nebula interface. 

To prepare the release of the smart-chip we will need to submit it to Google Marketplace. Let me know if you need any help with metadata texts for the package to be submitted.

**Comment by saumier** [comment]
@dev-aravind As we discussed in our phone call, please remove the search/reconiliation feature and add a note to the readme that it was removed on X date.

**Comment by dev-aravind** [comment]
@saumier [Steps](https://developers.google.com/workspace/marketplace/how-to-publish) to publish the smart chip

**Comment by saumier** [comment]
@fjjulien As per https://developers.google.com/workspace/marketplace/how-to-publish we need the following to complete the task of publishing to Google marketplace:

- [x] Privacy policy page
- [x] Terms of service page
- [x] Information on how users can get help


I create a single webpage in the Nebula repo. 
https://github.com/culturecreates/nebula/blob/main/doc/en/google-smart-chip.html.md

Can you help fill in the markdown?

**Comment by saumier** [comment]
@fjjulien I filled in the sections needed on the webpage for Google Smart-Chip with a first version.  Please feel free to edit this page as any point.  I have not translated to French. 

http://kg.artsdata.ca/doc/google-smart-chip

We will go ahead and submit the Artsdata Google Smart-Chip to the Google Marketplace using this first version of the texts.

**Comment by saumier** [comment]
@dev-aravind Please try to continue to submit our Artsdata Google Smart-Chip add-on to the Google Marketplace and let me know if you need my help.

**Comment by dev-aravind** [comment]
@saumier correct me if I'm wrong, the support email should be support@culturecreates.com and logo should be the artsdata favicon.

**Comment by fjjulien** [comment]
@saumier I reviewed and edited the [Nebula page](https://github.com/culturecreates/nebula/blob/main/doc/en/google-smart-chip.html.md).

**Comment by saumier** [comment]
> [@saumier](https://github.com/saumier) correct me if I'm wrong, the support email should be [support@culturecreates.com](mailto:support@culturecreates.com) and logo should be the artsdata favicon.

Actually the support email should be support@artsdata.ca

Yes for the logo.

**Comment by saumier** [comment]
Re-submitted App for Google review after splitting the privacy policy page from the homepage.

> Hello Google Developer,
> 
> Thank you for your patience while we reviewed your submission for project 175383630168(Project ID: artsdata-smart-chip). We need you to address the following items for us to continue your app’s verification:
> 
> The privacy policy URL you gave us (https://kg.artsdata.ca/doc/google-smart-chip) is the same as your homepage URL (https://kg.artsdata.ca/doc/google-smart-chip).
> Please take the following action(s) to continue with your request
> ⤷ Update and [resubmit your app](https://www.google.com/url?q=https://support.google.com/cloud/answer/9110914%23submit-app-ver%26zippy%3D%252Csteps-to-submit-your-app&sa=D&source=editors&ust=1689612709299105&usg=AOvVaw3M8JY3z51s8dy2ELfJMRtG) in Cloud Console with the following:
> 
> Homepage and privacy policy URLs that are not the same ([Learn more](https://support.google.com/cloud/answer/13806988?hl=en&ref_topic=13841839&sjid=3852194599686420982-NA#zippy=%2Cthe-privacy-policy-url-you-gave-us-is-the-same-as-your-homepage-url))

**Comment by saumier** [comment]
@dev-aravind I received this email saying that they approved our submission, but when I search Google market place for "Artsdata" nothing comes up. Any ideas of what to do next?

```
Hello Google Developer,

Thank you for your patience while we reviewed your submission. We've approved your OAuth App Verification request for project 175383630168(Project ID: artsdata-smart-chip) for brand verification.

You can check the latest status of your request on the OAuth Consent Screen of the [ Google API Console](https://console.developers.google.com/).
```

**Comment by dev-aravind** [comment]
@saumier this seems like an OAuth verification, did you submit the app for review in the [store listing page](https://console.cloud.google.com/apis/api/appsmarket-component.googleapis.com/googleapps_sdk_publish?inv=1&invt=AbzMyw&project=artsdata-smart-chip) during your setup? I submitted it right now.

**Comment by saumier** [comment]
Today's feedback

> Hi,
> 
> Your current App submission for Artsdata Smart-Chip (175383630168) has been rejected because it does not comply with some of the [Google Workspace Marketplace guidelines](https://developers.google.com/workspace/marketplace/about-app-review#areas) and has to be resubmitted.
> 
> In order to address the issues listed below, you will need to make changes in the [Google Workspace Marketplace SDK](https://console.cloud.google.com/apis/library/appsmarket-component.googleapis.com) settings in your app.
> 
> Give proper attribution when using Google Trademarks in your App name or Descriptions.
> Most of Google's products are trademarked. What this means for your product listing is that, whenever you use the names of these products, you must append a ™ symbol to them.
> 
> For example: Productivity Tool for Chat™
> 
> You may also add a footnote to the detailed description regarding the trademark.
> 
> For example: Google Chat™ is a trademark of Google LLC
> 
> You can find more information on our branding guidelines here:
> https://developers.google.com/workspace/marketplace/terms/branding#giving_proper_attribution
> 
> You can also find a list of trademarked Google products here:
> https://about.google/brand-resource-center/trademark-list/
> 
> Additional notes: Please ensure all Google products are correctly trademarked (using ™) within both the short and detailed descriptions.
> 
> Detailed description should provide a longer description of your app.
> Write your description to entice users to download your item. Make sure to start your description with a concise statement of what your item does, so users can understand the main idea and features at a glance. Your description can then go on to provide more detail but always limit yourself to describe the App itself and its functionality. Any additional detail should be published and available on your websites. Be sure your descriptions are not misleading, improperly formatted, non-descriptive, irrelevant, excessive, or inappropriate. Developers must provide clear and well-written short and detailed descriptions. We also don't allow unattributed or anonymous user testimonials in the product's description. Any issue with these considerations can be considered as a violation of the Spam & Placement in the Store Policy (https://developers.google.com/workspace/marketplace/terms/policies).
> 
> Additional notes: The detailed description should provide a more in-depth explanation of the app's functionality and usage.
> 
> Links for Privacy Policy, Terms of Service and Support are provided and point to the correct information.
> You can check additional info about this requirement here (https://developers.google.com/workspace/marketplace/create-listing#support_links).
> All provided links work and point to the correct information. For example, a link provided for a privacy policy points to a page describing the privacy policy.
> 
> Additional notes: Please ensure the Terms of Service and Privacy Policy links are functional and direct users to the correct pages.
> 
> The app is missing OAuth verification.
> Your app is requesting sensitive or restricted OAuth scopes that are not approved (https://support.google.com/cloud/answer/7454865). New users might not be able to install your application because the Sign-in with google is disabled.
> This happens when the scopes requested in your app code differ from scopes approved in your OAuth consent screen configuration page (https://console.cloud.google.com/apis/credentials/consent). The scopes configured in App script (https://developers.google.com/apps-script/concepts/scopes), Marketplace SDK (https://developers.google.com/workspace/marketplace/enable-configure-sdk#oauth_scopes), and Oauth consent screen (https://support.google.com/cloud/answer/10311615?hl=en&ref_topic=3473162) should be identical.
> Please add all sensitive/restricted scopes that the app is using to your GCP Console OAuth consent screen (https://console.cloud.google.com/apis/credentials/consent) configuration page and click “Submit for Verification"(https://developers.google.com/apps-script/guides/client-verification#requesting_verification).
> Please note that the Trust & Security team is responsible for the OAuth verification process.
> If your product is not yet available in the Marketplace make sure that in your Website you dont include buttons or links to the Marketplace until it is effectively approved. Unclickable links and coming soon texts can cause delays in the OAuth review process.
> Before re-submitting the add-on for the Google Workspace Marketplace, make sure your project has passed the OAuth verification process.
> 
> Additional notes: Google has not verified this app yet. If you don't verify your script project's OAuth client, users outside your domain see an unverified app screen when they try to authorize your script. You can request a verification of the OAuth client used by your app and its associated Cloud Platform (GCP) project. Once your app is verified, your users will no longer see the unverified app screen. https://support.google.com/cloud/answer/7454865#unverified-app-screen Please add all sensitive/restricted scopes that the app is using to your GCP Console OAuth consent screen (https://console.cloud.google.com/apis/credentials/consent) configuration page and click “Submit for Verification"(https://developers.google.com/apps-script/guides/client-verification#requesting_verification).
> Check this link for more information: https://developers.google.com/apps-script/guides/client-verification Please note that the Trust & Security team is responsible for the OAuth verification process.
> You can also follow the information in the following Blogpost for recommendations to undergo the OAuth verification process: https://developers.googleblog.com/2019/09/get-smart-about-preparing-your-app-for-OAuth-verfication.html
> 
> Please republish your App when an updated version is ready.
> 
> Thank you for your patience.
> 
> If you have any questions please contact [gwm-review@google.com](mailto:gwm-review@google.com). To get support, please refer to the [support section](https://developers.google.com/workspace/add-ons/workspace-support) of our documentation.
> 
> Thanks,
> 
> Google Workspace Marketplace Review Team

**Comment by dev-aravind** [comment]
@saumier It is not clear what went wrong, but I have updated the google trademarks, privacy policy and terms of service issues and resubmitted the app for review.

**Comment by saumier** [comment]
@dev-aravind Google marketplace sent another email at 9h21 AM ET to admin@culturecreates.com with reasons why it was rejected. Please take a look and try to fix. Thx.

**Comment by dev-aravind** [comment]
@saumier can you attach the email to the issue? I don't think that I'm part of the admin group. Thanks.

**Comment by saumier** [comment]
@dev-aravind I'll spend some time trying to understand what they are asking and let you know what specifics I can find. I'm setting this to low priority for now so we can focus on the other higher priority issues.

**Comment by saumier** [comment]
@kmdvs Please try to unblock this. The goal of the rollout is that any public user can use this Artsdata Smart Chip in their Google Excel sheet to roll over an Artsdata URI and get a popup with some details.  It is quite complicated to figure out if this needs to be sent to the Google marketplace (which we tried and received messages from Google that we need to make changes) and how a user can install it in their Google sheets (one installation should work with all sheets).  I can give you a demo so you understand the goal.

**Comment by kmdvs** [comment]
@saumier

1. I need permissions to get access to the artsdata-smart-chip googleapis.com project

I get
```
Troubleshooting info:
  Principal: k1m.d4v1s@gmail.com
  Resource: artsdata-smart-chip
  Troubleshooting URL: console.cloud.google.com/iam-admin/troubleshooter;permissions=resourcemanager.projects.get;principal=k1m.d4v1s@gmail.com;resources=%2F%2Fcloudresourcemanager.googleapis.com%2Fprojects%2Fartsdata-smart-chip/result

Missing or denied permissions:
  resourcemanager.projects.get
```

Or should I log in on a different account?

2. I need the last email sent, to troubleshoot from latest response

**Comment by saumier** [comment]
@kmdvs Can you work on this again. Instead of trying to find the email thread, you could use the service@culturecreates.com email to restart the process. I have given access with role "owner" to service@culturecreates.com.

https://console.cloud.google.com/iam-admin/iam?project=artsdata-smart-chip

**Comment by saumier** [comment]
@kmdvs Are you able to work on this?

**Comment by saumier** [comment]
@kmdvs Hi. Please confirm you see my message above and let me know if you are able to work on this?

**Comment by kmdvs** [comment]
@saumier Linked to [PR 177](https://github.com/culturecreates/nebula/pull/177)

**Comment by kmdvs** [comment]
Summary of changes made in the last month

- Worked on trademark / language issues --> resolved
- Worked on OAuth --> in progress: need video

[OAuth tasks.docx](https://github.com/user-attachments/files/24869661/OAuth.tasks.docx)

**Comment by saumier** [comment]
@kmdvs I question why we need so many scope authorizations from users. We are just reading a spreadsheet cell and if it is a URL from Artsdata.ca then we make an API call. Can you confirm that the scope is the minimal scope we need? Creating a video is crazy. I think we are getting to the point of abandoning the project if more work is still needed. We have a back log of more pressing issues.

**Comment by saumier** [comment]
@fjjulien Update: I have assigned my colleague, Kim Davis, to work on publishing the Artsdata smart chip to Google Marketplace. He have followed many requests from Google (see above). However, I am drawing the line regarding this latest request from Google to provide a video screen recording to reassure users that they can trust the app. 

I think there is a mis-match between what we are trying to accomplish with this little took, and the impression Google has about what is needed. It could be that we are asking for too many OAuth permissions, or that Google is asking for extreme measures.  This project is lost in the details. To revive it, we would need to take a step back and look at the big picture and try to understand why we are on the wrong path.  

Either way I am putting an end to this project as it is no longer providing any value.
